source/m3_math_utils.h

```diff
@@ -197,6 +197,17 @@ u64 rotr64(u64 n, unsigned c) {
     }                                                       \
     RES = (TYPE)A;
 
+#define OP_TRUNC_SAT(RES, A, TYPE, RMIN, RMAX)              \
+    if (UNLIKELY(isnan(A))) {                               \
+        return (TYPE)0;                                     \
```

Comment:
This and similar "return" lines below  should be `RES = ...`.  The `return` is for error conditions here.
Also, subsequent `if`'s should become `else if`.

Reply:
Good point. Will fix